bundles/org.openhab.binding.webexteams/src/main/java/org/openhab/binding/webexteams/internal/WebexTeamsHandler.java

```diff
@@ -0,0 +1,315 @@
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Collections.singletonList(WebexTeamsActions.class);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing...");
+        config = getConfigAs(WebexTeamsConfiguration.class);
+
+        updateStatus(this.status);
+
+        final String token = config.token;
+        final String clientId = config.clientId;
+        final String clientSecret = config.clientSecret;
+
+        if (!token.isBlank() && clientId.isBlank()) { // For bots
+            logger.debug("I'm a bot");
+        } else if (!clientId.isBlank()) { // For integrations
+            logger.debug("I'm a person.");
+            if (clientSecret.isEmpty()) {
+                this.status = ThingStatus.OFFLINE;
+                updateStatus(this.status, ThingStatusDetail.CONFIGURATION_ERROR, "@text/confErrorNoSecret");
+                return;
+            }
+        } else {
+            this.status = ThingStatus.OFFLINE;
+            updateStatus(this.status, ThingStatusDetail.CONFIGURATION_ERROR, "@text/confErrorTokenOrId");
+            return;
+        }
+
+        // background initialization:
+        scheduler.execute(() -> {
+            try {
+                this.client = new WebexTeamsApi(this, oAuthFactory, httpClient);
+                logger.debug("Trying to fetch account details");
+                Person p = this.client.getPerson();
+                logger.debug("Success: {}", p.getDisplayName());
+
+                this.status = ThingStatus.ONLINE;
+                updateStatus(this.status);
+
+                refresh();
+
+            } catch (NotAuthenticatedException e) {
+                logger.error("Failed to initialize client", e);
+                this.status = ThingStatus.OFFLINE;
+                updateStatus(this.status, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            }
+        });
+
+        // TODO: make interval configurable.
+        refreshService = scheduler.scheduleWithFixedDelay(this::refresh, 0, 300, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing...");
+        final ScheduledFuture<?> job = refreshService;
+        if (job != null) {
+            job.cancel(true);
+            refreshService = null;
+        }
+        this.client.dispose();
+    }
+
+    // mainly used to refresh the auth token when using OAuth
+    private void refresh() {
+        if (status == ThingStatus.ONLINE) {
+            Person person;
+            try {
+                person = client.getPerson();
+                updateState(CHANNEL_BOTNAME, StringType.valueOf(person.getDisplayName()));
+                updateProperty("personType", person.getType());
+                updateProperty("name", person.getDisplayName());
+
+                // Only when the identity is a person:
+                if (person.getType().equalsIgnoreCase("person")) {
```

Comment:
The good practice is to reverse: `"person".equalsIgnoreCase(person.getType())`

Reply:
Agreed.  Fixed.